test/Ocelot.AcceptanceTests/RoutingWithQueryStringTests.cs

```diff
@@ -62,7 +62,85 @@ public void should_return_response_200_with_query_string_template()
+                                    Port = port,
+                                },
+                            },
+                            UpstreamPathTemplate = "/api/units/{subscriptionId}/updates?unit={unitId}",
+                            UpstreamHttpMethod = new List<string> { "Get" },
+                        },
+                    },
+            };
+
+            this.Given(x => x.GivenThereIsAServiceRunningOn($"http://localhost:{port}", $"/api/subscriptions/{subscriptionId}/updates", $"?unitId={unitId}", 200, "Hello from Laura"))
+                .And(x => _steps.GivenThereIsAConfiguration(configuration))
+                .And(x => _steps.GivenOcelotIsRunning())
+                .When(x => _steps.WhenIGetUrlOnTheApiGateway($"/api/units/{subscriptionId}/updates?unit={unitId}"))
+                .Then(x => _steps.ThenTheStatusCodeShouldBe(HttpStatusCode.OK))
+                .And(x => _steps.ThenTheResponseBodyShouldBe("Hello from Laura"))
+                .BDDfy();
+        }
+
+
+        [Fact]
+        public void Should_return_response_200_with_query_string_template_additional_key() //issue #327
+        {
+            var subscriptionId = Guid.NewGuid().ToString();
+            var unitId = Guid.NewGuid().ToString();
+            var port = RandomPortFinder.GetRandomPort();
+
+            var configuration = new FileConfiguration
+            {
+                Routes = new List<FileRoute>
+                    {
+                        new()
+                        {
+                            DownstreamPathTemplate = "/api/subscriptions/{subscriptionId}/updates?unitId={unitId}",
+                            DownstreamScheme = "http",
+                            DownstreamHostAndPorts = new List<FileHostAndPort>
+                            {
+                                new()
+                                {
+                                    Host = "localhost",
+                                    Port = port,
+                                },
+                            },
+                            UpstreamPathTemplate = "/api/units/{subscriptionId}/updates?unit={unitId}",
+                            UpstreamHttpMethod = new List<string> { "Get" },
+                        },
+                    },
+            };
+
+            this.Given(x => x.GivenThereIsAServiceRunningOn($"http://localhost:{port}", $"/api/subscriptions/{subscriptionId}/updates", $"?unitId={unitId}&x=y", 200, "Hello from Laura"))
+                .And(x => _steps.GivenThereIsAConfiguration(configuration))
+                .And(x => _steps.GivenOcelotIsRunning())
+                .When(x => _steps.WhenIGetUrlOnTheApiGateway($"/api/units/{subscriptionId}/updates?unit={unitId}&x=y"))
+                .Then(x => _steps.ThenTheStatusCodeShouldBe(HttpStatusCode.OK))
+                .And(x => _steps.ThenTheResponseBodyShouldBe("Hello from Laura"))
+                .BDDfy();
+        }

```

Comment:
I see a copy-paste code! So, DRY violation has detected!

You can extract the body code and define a private helper with one string parameter
```csharp
private void ShouldReturnResponse200WithQueryStringTemplate(string queryUrl)
{
    // ...
    .When(x => _steps.WhenIGetUrlOnTheApiGateway(queryUrl))
    // ...
}
```
and reuse this private helper for both fact-methods varying the param like that:
```csharp
public void Should_return_response_200_with_query_string_template_different_keys()
{
    // ...
    ShouldReturnResponse200WithQueryStringTemplate($"/api/units/{subscriptionId}/updates?unit={unitId}"));
}

public void Should_return_response_200_with_query_string_template_additional_key()
{
    // ...
    ShouldReturnResponse200WithQueryStringTemplate($"/api/units/{subscriptionId}/updates?unit={unitId}&x=y"));
}
```
Finally, it is better to define xUnit Theory. 😄